arc-mlir/src/include/arc/arc-ops.td

```diff
@@ -193,13 +193,24 @@ def MakeTuple
   let results = (outs AnyTuple);
 
   let verifier = [{
-    unsigned NumOperands = this->getOperation()->getNumOperands();
-    auto TupleTy =
-        this->getOperation()->getResult(0)->getType().dyn_cast<TupleType>();
+    auto Operation = this->getOperation();
+    auto OperandTys = Operation->getOperands().getTypes();
+    unsigned NumOperands = Operation->getNumOperands();
+    auto TupleTy = Operation->getResult(0).getType().dyn_cast<TupleType>();
+    auto ElemTys = TupleTy.getTypes();
     if (NumOperands != TupleTy.size())
       return emitOpError("result does not match the number of operands: found ")
              << NumOperands << " but expected " << TupleTy.size()
              << " operands";
+    auto a = ElemTys.begin();
+    auto b = OperandTys.begin();
+    while (a != ElemTys.end() and b != OperandTys.end()) {
```

Comment:
C++ is strange, how did `and` go through the compiler?

I'd fix the compilation error by doing (this is untested):

```C++
    auto Operation = this->getOperation();
    unsigned NumOperands = Operation->getNumOperands();
    TupleType TupleTy = Operation->getResult(0).getType().cast<TupleType>();
    auto ElemTys = TupleTy.getTypes();
    if (NumOperands != TupleTy.size())
      return emitOpError("result does not match the number of operands: found ")
             << NumOperands << " but expected " << TupleTy.size()
             << " operands";
    unsigned I = 0;
    for (const Type &ElemTy : Operation->getOperands().getTypes()) {
        if (ElemTys[I] != ElemTy)
           return emitOpError("Operand types do not match, found ")
              << ElemTy << " but expected " << ElemTys[I];
        I++;
    }
    return mlir::success();

```